sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdToken.java

```diff
@@ -73,17 +74,26 @@ public Object getValue() {
             return codec.defaultValue();
         }
 
+        if (this.hasConvertedValue) {
+            return this.value;
+        }
+
         if (this.value instanceof ByteBuf) {
             final ByteBuf buffer = (ByteBuf) this.value;
+            buffer.markReaderIndex();
```

Comment:
Why do we need to mark the reader index? Do we expect to revisit the same `RntbdToken` instance?

Reply:
Valid point - this is not needed anymore because I store the flag about conversion happening now. Will revert that part of the change.